### PR TITLE
Fix use of floating number for the `timeout` and `forceKillAfterTimeout` options

### PR DESCRIPTION
**Pull request by Kikobeats**
When the `timeout` or `forceKillAfterTimeout` a float number is causes an error that can be avoided

### CODE REVIEW

**Comment by Kikobeats** [comment]
Ok I can see it's an intentional behavior

```
   64:   test('`forceKillAfterTimeout` should not be a float', t => {    
```

not sure why 🤔

**Comment by sindresorhus** [comment]
I don't remember. Maybe try Git blame.

**Comment by Kikobeats** [comment]
the code was refactored several times but `isInteger` was introduced here:
https://github.com/sindresorhus/execa/commit/0bd5596bfab81f67de44545adeb3d6c912a0db85#diff-168726dbe96b3ce427e7fedce31bb0bcR246

I couldn't determinate if number over float is necessary for a thing, maybe @ehmicky can know 🙂

**Comment by ehmicky** [comment]
Yes, I think we should allow floats.
However we should still disalllow `NaN` and `Infinity`, so `Number.isFinite()` might be better than `typeof value === 'number'` I believe?

**Comment by sindresorhus** [comment]
Yes, let’s do that.

File: lib/kill.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const getForceKillAfterTimeout = ({forceKillAfterTimeout = true}) => {
 		return DEFAULT_FORCE_KILL_TIMEOUT;
 	}
 
-	if (!Number.isInteger(forceKillAfterTimeout) || forceKillAfterTimeout < 0) {
+	if (Number.isFinite(forceKillAfterTimeout) || forceKillAfterTimeout < 0) {
```
Comment (sindresorhus) [inline lib/kill.js:47]:
```suggestion
	if (!Number.isFinite(forceKillAfterTimeout) || forceKillAfterTimeout < 0) {
```

File: lib/kill.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ const setupTimeout = (spawned, {timeout, killSignal = 'SIGTERM'}, spawnedPromise
 		return spawnedPromise;
 	}
 
-	if (!Number.isInteger(timeout) || timeout < 0) {
+	if (Number.isFinite(number) || timeout < 0) {
```
Comment (ehmicky) [inline lib/kill.js:74]:
```suggestion
	if (!Number.isFinite(number) || timeout < 0) {
```

File: lib/kill.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ const setupTimeout = (spawned, {timeout, killSignal = 'SIGTERM'}, spawnedPromise
 		return spawnedPromise;
 	}
 
-	if (!Number.isInteger(timeout) || timeout < 0) {
+	if (!Number.isFinite(number) || timeout < 0) {
```
Comment (ehmicky) [inline lib/kill.js:74]:
```suggestion
	if (!Number.isFinite(timeout) || timeout < 0) {
```

**Comment by ehmicky** [comment]
The CI test are failing due to the following test: 

https://github.com/sindresorhus/execa/blob/a827d82203a1440e585276bef5d399a5953801f1/test/kill.js#L64

This probably should be changed to check for `NaN` instead?

**Comment by Kikobeats** [comment]
@ehmicky test updated; still, some tests are failing  but for a different reason no related with this PR

**Comment by ehmicky** [comment]
Thanks @Kikobeats,

It looks like you accidentally also removed the test that checks that the argument is not negative. Could you please add it back? Thanks :)

**Comment by Kikobeats** [comment]
@ehmicky ops, done!
